hack/runtime-migrator/cmd/migration/migration.go

```diff
@@ -83,60 +81,60 @@ func (m Migration) Do(ctx context.Context, runtimeIDs []string) error {
 	}
 
 	run := func(runtimeID string) {
-		shoot, err := m.fetchShoot(ctx, shootList, m.shootClient, runtimeID)
+		shootToMigrate, err := shoot.Fetch(ctx, shootList, m.shootClient, runtimeID)
 		if err != nil {
 			reportError(runtimeID, "", "Failed to fetch shoot", err)
 			return
 		}
 
-		if shootIsBeingDeleted(shoot) {
-			reportError(runtimeID, shoot.Name, "Runtime is being deleted", nil)
+		if shoot.IsBeingDeleted(shootToMigrate) {
+			reportError(runtimeID, shootToMigrate.Name, "Runtime is being deleted", nil)
 			return
 		}
 
-		migrationCtx, cancel := context.WithTimeout(ctx, timeoutK8sOperation)
+		migrationCtx, cancel := context.WithTimeout(ctx, config2.TimeoutK8sOperation)
 		defer cancel()
 
-		runtime, err := m.runtimeMigrator.Do(migrationCtx, *shoot)
+		runtime, err := m.runtimeMigrator.Do(migrationCtx, *shootToMigrate)
 
 		if err != nil {
-			reportError(runtimeID, shoot.Name, "Failed to migrate runtime", err)
+			reportError(runtimeID, shootToMigrate.Name, "Failed to migrate runtime", err)
 			return
 		}
 
 		err = m.outputWriter.SaveRuntimeCR(runtime)
 		if err != nil {
-			reportError(runtimeID, shoot.Name, "Failed to save runtime CR", err)
+			reportError(runtimeID, shootToMigrate.Name, "Failed to save runtime CR", err)
 			return
 		}
 
-		shootComparisonResult, err := m.runtimeVerifier.Do(runtime, *shoot)
+		shootComparisonResult, err := m.runtimeVerifier.Do(runtime, *shootToMigrate)
 		if err != nil {
-			reportValidationError(runtimeID, shoot.Name, "Failed to verify runtime", err)
+			reportValidationError(runtimeID, shootToMigrate.Name, "Failed to verify runtime", err)
 			return
 		}
 
 		if !shootComparisonResult.IsEqual() {
 			err = m.outputWriter.SaveComparisonResult(shootComparisonResult)
 			if err != nil {
-				reportError(runtimeID, shoot.Name, "Failed to save comparison result", err)
+				reportError(runtimeID, shootToMigrate.Name, "Failed to save comparison result", err)
 				return
 			}
 
-			reportUnwantedUpdateDetected(runtimeID, shoot.Name, "Runtime CR can cause unwanted update in Gardener")
+			reportUnwantedUpdateDetected(runtimeID, shootToMigrate.Name, "Runtime CR can cause unwanted update in Gardener")
 			return
 		}
 
 		if m.isDryRun {
```

Comment:
Since we are changing  the code in this area, please consider refactoring this part by reducing if-else just to if. This could improve code readability by removing nested code sections.